auditlog/admin.py

```diff
@@ -40,6 +40,11 @@ def has_change_permission(self, request, obj=None):
         return False
 
     def has_delete_permission(self, request, obj=None):
+        if request.resolver_match and request.resolver_match.url_name not in [
+            pattern.name for pattern in self.urls if pattern.name
```

Comment:
can't we move the admin urls patterns list creation out of this method to not calculate it every time?

Reply:
I'm not sure when in the lifecycle of the admin `self.urls` is available, so I'll just move this expression into a cached property.

Reply:
Done.